modules/core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/marshallers/json/document/DocumentPropertiesJsonReader.java

```diff
@@ -199,14 +205,17 @@ private Object getScalarPropertyValue(Property property, JsonNode jn, Type type)
                         return blob.getString();
                     }
                 }
-                log.info("Unable to parse the property {}", property::getXPath);
-                return null;
+                throw new MarshallingException("Unable to parse the property " +  property.getXPath());
             }
             Object object = null;
             for (Class<?> clazz : resolver.getManagedClasses()) {
-                object = readEntity(clazz, clazz, jn);
-                if (object != null) {
-                    break;
+                try {
+                    object = readEntity(clazz, clazz, jn);
+                    if (object != null) {
+                        break;
+                    }
+                } catch (Exception e) {
+                    log.info("Unable to read the entity - {}", e.getMessage(), e);
```

Comment:
Can we use `e::getMessage` instead?

Reply:
AFAIK there's no API allowing to mix Supplier and throwable. Regarding the [Javadoc](https://logging.apache.org/log4j/2.x/log4j-api/apidocs/org/apache/logging/log4j/Logger.html#info-java.lang.String-org.apache.logging.log4j.util.Supplier...-) Supplier seems to be for parameters only.

Otherwise, the following is correct:
```java
log.info(() -> String.format("Unable to read the entity - %s", e.getMessage()), e);
```
Don't really know if it's beneficial regarding verbosity and code reading, wdyt?

Reply:
Ah yes I always forget if it's valid or not. No then keep your old code, it doesn't matter for an INFO message that will have to be generated to go to Datadog anyway.

Reply:
We could also do the following, we have some occurrences in the code:
```
log.info("Unable to read the entity - {}", e::getMessage, () -> e);
```

Reply:
Didn't know that, many thanks. Let's do that!